File: 2125. Number of Laser Beams in a Bank/solution.cpp
class Solution {
public:
    int numberOfBeams(vector<string>& bank) {
        vector<int> arr;

        // Count the number of "beams" in each string
        for (int i = 0; i < bank.size(); ++i) {
            int beams = count(bank[i].begin(), bank[i].end(), '1');
            if (beams > 0) {
                arr.push_back(beams);
            }
        }

        // If there are less than 2 "beams," return 0
        if (arr.size() < 2) {
            return 0;
        }

        // Calculate the sum using a loop
        int sum_result = 0;
        for (int i = 0; i < arr.size() - 1; ++i) {
            sum_result += arr[i] * arr[i + 1];
        }

        return sum_result;
    }
};

File: 2125. Number of Laser Beams in a Bank/solution.py
class Solution:
    def numberOfBeams(self, bank: List[str]) -> int:
        arr = []
        # Count the number of "beams" in each string
        for i in range(len(bank)):
            beams = bank[i].count('1')
            if beams > 0:
                arr.append(beams)
                
        # *** NOTE: beams_count = [s.count('1') for s in bank if '1' in s] is a more efficient way to do this ***

        # If there are less than 2 "beams," return 0
        if len(arr) < 2:
            return 0

        # Calculate the sum using a list comprehension
        sum_result = sum(arr[i] * arr[i + 1] for i in range(len(arr) - 1))

        return sum_result